rust/cbork-utils/src/decode_helper.rs

```diff
@@ -92,6 +92,21 @@ pub fn decode_any<'d>(d: &mut Decoder<'d>, from: &str) -> Result<&'d [u8], decod
     Ok(bytes)
 }
 
+/// Extracts the raw bytes of a CBOR map from a decoder based on specified positions.
+/// This function retrieves the raw byte representation of a CBOR map between the given
+/// start and end positions from the decoder's underlying buffer.
+///
+/// # Errors
+///  - Invalid map byte range: indices out of bounds
+pub fn get_bytes(
+    d: &Decoder<'_>, map_start: usize, map_end: usize,
+) -> Result<Vec<u8>, decode::Error> {
```

Comment:
Maybe we can skip a construction of the vector here and let the caller decide if the vector is needed or not?

Something like this should work:
```rust
pub fn get_bytes<'a>(
    d: &Decoder<'a>, map_start: usize, map_end: usize,
) -> Result<&'a [u8], decode::Error> {
    d.input()
        .get(map_start..map_end)
        .ok_or_else(|| decode::Error::message("Invalid map byte range: indices out of bounds"))
}
```

Reply:
Updated